### PR TITLE
Enhance error messages for failed syscalls

### PR DESCRIPTION
**Pull request by dominiklohmann**

### CODE REVIEW

File: libvast/src/detail/posix.cpp [hunk 1]
```diff
@@ -226,33 +223,41 @@ bool poll(int fd, int usec) {
   if (rc < 0) {
     switch (rc) {
       default:
-        VAST_RAISE_ERROR(std::logic_error, "unhandled select() error");
+        VAST_RAISE_ERROR(std::logic_error, "unhandled select(2) error");
```
Comment (mavam) [inline libvast/src/detail/posix.cpp:226]: Can we return an error here as well now?

File: libvast_test/vast/test/fixtures/filesystem.hpp [hunk 2]
```diff
@@ -21,12 +23,12 @@ struct filesystem {
   filesystem() {
     // Fresh afresh.
     rm(directory);
-    mkdir(directory);
+    if (auto err = mkdir(directory))
+      // error is non-recoverable, so we just abort
+      vast::die(vast::render(err));
```
Comment (mavam) [inline libvast_test/vast/test/fixtures/filesystem.hpp:28]:
```suggestion
      FAIL(vast::render(err));
```

File: libvast_test/vast/test/fixtures/filesystem.hpp [hunk 3]
```diff
@@ -13,6 +13,8 @@
 
 #pragma once
 
+#include "vast/die.hpp"
```
Comment (mavam) [inline libvast_test/vast/test/fixtures/filesystem.hpp:16]:
```suggestion
```

**Review by mavam — CHANGES_REQUESTED** [review]
LGTM.

Just some small stuff.

File: libvast/src/detail/posix.cpp (hunk 1 above)
Comment (dominiklohmann) [reply libvast/src/detail/posix.cpp:226]: I'd rather call `vast::die` here.

**Review by mavam — APPROVED** [review]

**Pull request merged**